cpp/test/mesh/rectangle.cpp

```diff
@@ -0,0 +1,211 @@
+// Copyright (C) 2024 Paul Kühner
+//
+// This file is part of DOLFINX (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#include <mpi.h>
+
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/matchers/catch_matchers.hpp>
+#include <catch2/matchers/catch_matchers_floating_point.hpp>
+#include <catch2/matchers/catch_matchers_range_equals.hpp>
+#include <catch2/matchers/catch_matchers_vector.hpp>
+
+#include <dolfinx/common/MPI.h>
+#include <dolfinx/graph/AdjacencyList.h>
+#include <dolfinx/mesh/Mesh.h>
+#include <dolfinx/mesh/generation.h>
+#include <dolfinx/mesh/utils.h>
+#include <dolfinx/refinement/interval.h>
+#include <dolfinx/refinement/refine.h>
+
+#include "util.h"
+
+using namespace dolfinx;
+using namespace Catch::Matchers;
+
+TEMPLATE_TEST_CASE("Rectangle quadrilateral mesh",
+                   "[mesh][rectangle][quadrilateral]", float, double)
+{
+  using T = TestType;
```

Comment:
Where is `TestType` defined?

Reply:
Is part of the catch macro `TEMPLATE_TEST_CASE`. Here, its either `float` or `double`.